enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,264 @@
+// RotationalAndNonRotational
+type DeviceMechanicalProperty string
+
+const (
+    // The mechanical properties of the devices
+    // Rotational refers to magnetic disks
+    Rotational DeviceMechanicalProperty = "rotational"
+    // NonRotational refers to ssds
+    NonRotational DeviceMechanicalProperty = "nonRotational"
+    // RotationalAndNonRotational refers to both magnetic and ssds
+    RotationalAndNonRotational DeviceMechanicalProperty = "rotationalAndNonRotational"
+)
+
+type DeviceDiscoveryPolicyType string
+
+const (
+    // The DeviceDiscoveryPolicies that will be supported by the LSO.
+    // These Discovery policies will based on lsblk's type output
+    // Disk represents a device-type of disk
+    Disk DeviceDiscoveryPolicyType = "disk"
+    // Part represents a device-type of partion
+    Part DeviceDiscoveryPolicyType = "part"
+)
+
+type DeviceInclusionSpec struct {
+    // DeviceTypes that should be used for automatic detection. This would be one of the types supported
+    // by the local-storage operator. Currently the supported types are: disk,part
+    // If the list is empty the default value will be `[disk]`.
+    DeviceTypes []DeviceDiscoveryPolicyType `json:"deviceType"`
+
+    // DeviceMechanicalProperty denotes whether Rotational or NonRotational disks should be used.
+    // by default, RotationalAndNonRotational which matches all disks.
+    // +optional
+    DeviceMechanicalProperty DeviceMechanicalProperty `json:"deviceMechanicalProperty"`
+
+    // MinSize is the minimum size of the device which needs to be included
+    // +optional
+    MinSize resource.Quantity `json:"minSize"`
+
+    // MaxSize is the maximum size of the device which needs to be included
+    // +optional
+    MaxSize resource.Quantity `json:"maxSize"`
+
+    // Models is a list of device models. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Models []string `json:"models"`
+    // Vendors is a list of device vendors. If not empty, the device's model as outputted by lsblk needs
+    // to contain at least one of these strings.
+    // +optional
+    Vendors []string `json:"vendors"`
+}
+```
+
+An example of autoDetectVolume CR instance:
+
+```yaml
+apiVersion: local.storage.openshift.io/v1
+kind: AutoDetectVolume
+metadata:
+  name: example-autodetect
+spec:
+  nodeSelector:
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: kubernetes.io/hostname
+        operator: In
+          values:
+          - worker-0
+          - worker-1
+  deviceDiscoverPolicies:
+    - fsType: ext4
+      storageClassName: example-storageclass1
+      volumeMode: filesystem
+      minDeviceCount: 5
+      deviceInclusionSpec:
+        deviceTypes:
+        - part
+        deviceMechanicalProperty: nonRotational
+        minSize: 10G
+        maxSize: 100G
+    - storageClassName: example-storageclass2
```

Comment:
Multiple policies in the same CR is confusing to me. These are independent policies, right?

Reply:
yes these are independent policies.